src/clio/main/Application.h

```diff
@@ -0,0 +1,229 @@
+#ifndef CLIO_APPLICATION_H
+#define CLIO_APPLICATION_H
+
+#include <memory>
+#include <optional>
+
+#include <boost/asio.hpp>
+#include <boost/asio/ssl.hpp>
+
+#include <clio/main/Config.h>
+
+namespace RPC {
+class Counters;
+class WorkQueue;
+}  // namespace RPC
+
+namespace Backend {
+class BackendInterface;
+}
+
+class DOSGuard;
+class NetworkValidatedLedgers;
+class ReportingETL;
+class ETLLoadBalancer;
+class SubscriptionManager;
+
+class HttpSession;
+class SslHttpSession;
+
+template <class PlainSession, class SslSession>
+class Listener;
+
+namespace Server {
+using HttpServer = Listener<HttpSession, SslHttpSession>;
+}
+
+class Application
+{
+public:
+    virtual ~Application(){};
+
+    virtual Config const&
+    config() const = 0;
+
+    virtual boost::asio::io_context&
+    rpc() const = 0;
+
+    virtual boost::asio::io_context&
+    etl() const = 0;
+
+    virtual RPC::Counters&
+    counters() const = 0;
+
+    virtual std::optional<boost::asio::ssl::context>&
+    sslContext() const = 0;
+
+    virtual ETLLoadBalancer&
+    balancer() const = 0;
+
+    virtual SubscriptionManager&
+    subscriptions() const = 0;
+
+    virtual Backend::BackendInterface&
+    backend() const = 0;
+
+    virtual NetworkValidatedLedgers&
+    ledgers() const = 0;
+
+    virtual DOSGuard&
+    dosGuard() const = 0;
+
+    virtual ReportingETL&
+    reporting() const = 0;
```

Comment:
Needs a better name, we might as well just change the name of `ReportingETL` entirely actually.